### ISSUE TITLE
Add Frequency Reference Options for Simulation

### ISSUE THREAD

**Issue by rodrigomha** [NREL-SIIP/PowerSimulationsDynamics.jl#60]
Including Center-of-Inertia for system frequency, or using Frequency-Divider approaches to obtain frequency for each bus that can be passed directly to inverters (skipping PLL implementation).

**Comment by m-bossart** [comment]
@jd-lara Would this change also allow for using a device as the frequency ` ReferenceBus` that doesn't have frequency as a dynamic state (Droop GFM)?

**Comment by jd-lara** [comment]
We can account for that as an option. This is an interesting question about how to implement it. We can add another option.

**Comment by m-bossart** [comment]
Yeah it seems like the current implementation is limited to devices that have frequency as a state that is therefore already stored in the result.
